corehq/apps/app_manager/models.py

```diff
@@ -2989,6 +2991,108 @@ def update_app_case_meta(self, app_case_meta):
                 meta.add_closer(self.unique_id, action.close_condition)
 
 
+class ShadowForm(AdvancedForm):
+    form_type = 'shadow_form'
+    # The unqiue id of the form we are shadowing
+    shadow_parent_form_id = StringProperty(required=False, default=None)
+
+    # form actions to be merged with the parent actions
+    extra_actions = SchemaProperty(AdvancedFormActions)
+
+    @property
+    def shadow_parent_form(self):
+        if not self.shadow_parent_form_id:
+            return None
+        else:
+            app = self.get_app()
+            return app.get_form(self.shadow_parent_form_id)
+
+    @property
+    def source(self):
+        if self.shadow_parent_form_id:
+            return self.shadow_parent_form.source
+        from corehq.apps.app_manager.views.utils import get_blank_form_xml
+        return get_blank_form_xml("Untitled Form")
```

Comment:
I think I should actually just remove that string, as this empty form xml should never be rendered or shown to the user. (It's just a placeholder until the user selects a shadow parent)